access.go

```diff
@@ -162,36 +162,53 @@ func (s *Server) HandleAccessRequest(w *Response, r *http.Request) *AccessReques
 }
 
 func (s *Server) handleAuthorizationCodeRequest(w *Response, r *http.Request) *AccessRequest {
-	auth, err := CheckBasicAuth(r)
+	clientID := r.Form.Get("client_id")
+	if clientID == "" {
+		w.SetError(E_UNAUTHORIZED_CLIENT, "")
+		return nil
+	}
+	client, err := w.Storage.GetClient(clientID)
 	if err != nil {
 		w.SetError(E_SERVER_ERROR, "")
 		w.InternalError = err
 		return nil
 	}
+	if client == nil {
+		w.SetError(E_UNAUTHORIZED_CLIENT, "")
+		return nil
+	}
+	var publicClient bool
+	if CheckClientSecret(client, ""){
+		publicClient=true
 
-	var clientID string
-	var client Client
-	if auth == nil {
-		clientID = r.Form.Get("client_id")
-		if clientID == "" {
-			w.SetError(E_UNAUTHORIZED_CLIENT, "")
-			return nil
-		}
-		client = getClientWithoutSecret(clientID, w.Storage, w)
-	} else {
+	} else{
 		// get client authentication
 		auth := GetClientAuth(w, r, s.Config.AllowClientSecretInParams)
 		if auth == nil {
 			return nil
 		}
-		client = getClient(auth, w.Storage, w)
+		if !CheckClientSecret(client,auth.Password){
+			w.SetError(E_UNAUTHORIZED_CLIENT, "")
+			return nil
+		}
+	}
+
+
+	var codeVerifier string
+	// Optional PKCE support (https://tools.ietf.org/html/rfc7636)
+	if codeVerifier = r.Form.Get("code_verifier"); len(codeVerifier) == 0 {
+		if s.Config.RequirePKCEForPublicClients && publicClient {
```

Comment:
Just curious, is `RequirePKCEForPublicClients` configurable? What if it's set to false?

Reply:
it will not check the PKCE challenge / verifier if it's not enabled